src/m365/teams/commands/chat/chat-member-remove.spec.ts

```diff
@@ -0,0 +1,229 @@
+    const deleteStub = sinon.stub(request, 'delete').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/chats/${chatId}/members/${chatMemberId}`) {
+        return;
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { chatId: chatId, id: chatMemberId, confirm: true, verbose: true } });
+    assert(deleteStub.called);
+  });
+
+
+  it('removes the specified member retrieved by user principal name when prompt confirmed', async () => {
+    sinon.stub(odata, 'getAllItems').callsFake(async url => {
+      if (url === `https://graph.microsoft.com/v1.0/chats/${chatId}/members`) {
+        return chatMembers;
+      }
+
+      throw 'Invalid request';
+    });
+    const deleteStub = sinon.stub(request, 'delete').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/chats/${chatId}/members/${chatMemberId}`) {
+        return;
+      }
+
+      throw 'Invalid request';
+    });
+
+    sinonUtil.restore(Cli.prompt);
+    sinon.stub(Cli, 'prompt').callsFake(async () => (
+      { continue: true }
+    ));
+
+    await command.action(logger, { options: { chatId: chatId, userName: userPrincipalName, verbose: true } });
+    assert(deleteStub.called);
+  });
+
+  it('throws error when member by specifying userName is not found in the chat', async () => {
+    sinon.stub(odata, 'getAllItems').callsFake(async url => {
+      if (url === `https://graph.microsoft.com/v1.0/chats/${chatId}/members`) {
+        return [...chatMembers.slice(1)];
+      }
+
+      throw 'Invalid request';
+    });
+
+    await assert.rejects(command.action(logger, { options: { chatId: chatId, userName: userPrincipalName, confirm: true, verbose: true } }), new CommandError(`Member with userName ${userPrincipalName} could not be found in the chat`));
+  });
+
+  it('throws error when member by specifying userId is not found in the chat', async () => {
+    sinon.stub(odata, 'getAllItems').callsFake(async url => {
+      if (url === `https://graph.microsoft.com/v1.0/chats/${chatId}/members`) {
+        return [...chatMembers.slice(1)];
+      }
+
+      throw 'Invalid request';
+    });
+
+    await assert.rejects(command.action(logger, { options: { chatId: chatId, userId: userId, confirm: true, verbose: true } }), new CommandError(`Member with userId ${userId} could not be found in the chat`));
+  });
+
+  it('prompts before removing the specified message when confirm option not passed', async () => {
+    await command.action(logger, { options: { chatId: chatId, id: chatMemberId } });
+
+    let promptIssued = false;
+
+    if (promptOptions && promptOptions.type === 'confirm') {
+      promptIssued = true;
+    }
+
+    assert(promptIssued);
+  });
+
+  it('aborts removing the specified chat member when confirm option not passed and prompt not confirmed', async () => {
+    const deleteSpy = sinon.spy(request, 'delete');
```

Comment:
Let's create a stub for this test. While it's not really needed to make this test succeed, imagine that we break this test and a request is actually executed. Since we don't stub it, we'll get a 10s timeout error. Meanwhile when we stub this request, we will get an error saying `assert(deleteSpy.notCalled);` failed.